app/test/task/end2end_test.dart

```diff
@@ -18,7 +18,7 @@ import '../shared/test_services.dart';
 const String goldenDir = 'test/task/testdata/goldens';
 
 // TODO: generalize golden testing, use env var for regenerating all goldens.
```

Comment:
I'll try to adopt something similar in a follow-up PR. Am I right that this will update the files locally, but fail in CI if they are different?

Reply:
There are several nifty features:

* It can write a golden file with several sections with expectations. And tell what section is bad
* It chooses the filenames from the name of the test_file (by dissecting the snapshot)
* It can regenerate a missing file (unless on CI), It will update files given an environment variable.